integrations/fallback_chat/src/haystack_integrations/components/generators/fallback_chat/chat_generator.py

```diff
@@ -0,0 +1,404 @@
+from concurrent.futures import TimeoutError as FuturesTimeoutError
+from threading import Event as ThreadEvent
+from typing import Any
+
+from haystack import component, default_from_dict, default_to_dict
+from haystack.components.generators.chat.types import ChatGenerator
+from haystack.dataclasses import ChatMessage, StreamingCallbackT
+from haystack.tools import Tool, Toolset
+from haystack.utils.deserialization import deserialize_component_inplace
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class FallbackChatGenerator:
+    """
+    A chat generator wrapper that tries multiple chat generators sequentially with a per-generator timeout.
+
+    It forwards all parameters transparently to the underlying generators and returns the first successful result.
+    If all generators fail or time out, it raises a RuntimeError with details.
+
+    The timeout behavior respects individual generator timeouts:
+    - If a generator has its own timeout setting, the effective timeout is the minimum of the FallbackChatGenerator
+      timeout and the generator's timeout
+    - If a generator has no timeout setting, the FallbackChatGenerator timeout is used
+    - For non-streaming calls, the timeout is applied to the complete response
+    - For streaming calls, the timeout is applied as a time-to-first-token (TTFT) deadline
+
+    Failover is automatically triggered for these error types:
+    - 429 Rate limit errors
+    - 401 Authentication errors
+    - 400 Context length errors
+    - 408 Timeout errors
+    - 500+ Server errors
+    - Any other exception raised by a generator
+    """
+
+    def __init__(
+        self,
+        generators: list[ChatGenerator],
+        timeout: float = 8.0,
+    ):
+        """
+        :param generators: A non-empty list of chat generator components to try in order.
+        :param timeout: Per-generator timeout in seconds. Must be > 0.
+                       This acts as an upper bound for individual generator timeouts.
+                       For non-streaming calls, this is the completion timeout.
+                       For streaming calls, this is the time-to-first-token (TTFT) timeout.
+                       If individual generators have their own timeout settings, the effective
+                       timeout will be the minimum of this value and the generator's timeout.
+        """
+        if not generators:
+            msg = "'generators' must be a non-empty list"
+            raise ValueError(msg)
+        if timeout is None or timeout <= 0:
+            msg = "'timeout' must be a positive number of seconds"
+            raise ValueError(msg)
+
+        # Validation via duck-typing: require a callable 'run' method
+        for gen in generators:
+            if not hasattr(gen, "run") or not callable(gen.run):
+                msg = "All items in 'generators' must expose a callable 'run' method (duck-typed ChatGenerator)"
+                raise TypeError(msg)
```

Comment:
I'm not sure this check is needed. At least in our other components that take in components in their init we don't strictly double check that they are a Haystack component.